### PR TITLE
AC Convert Test - LPD-30748 | master

### PR DESCRIPTION
**Pull request by ViniLopes20**
https://liferay.atlassian.net/browse/LPD-30748

### CODE REVIEW

**Comment by ViniLopes20** [comment]
ci:test:analytics-cloud

**Comment by ViniLopes20** [comment]
ci:test:analytics-cloud

**Comment by ViniLopes20** [comment]
ci:test:sf

**Comment by ViniLopes20** [comment]
ci:test:relevant

**Comment by ViniLopes20** [comment]
ci:test:sf

File: modules/test/playwright/tests/osb-faro-web/segments.spec.ts [hunk 1]
```diff
@@ -362,3 +368,264 @@ test('Search the Segment Profile Distribution', async ({apiHelpers, page}) => {
...
+	apiHelpers,
+	page,
+}) => {
+	const anonymousIdentityID = '87';
+	const date1 = new Date();
+	const pageName = 'Liferay - AC Page';
+	const knownIndividualName = 'ac';
+	const dynamicSegmentName = 'Test Dynamic Segment';
+	const staticSegmentName = 'Test Static Segment';
+	const channelName = 'My Property - ' + getRandomString();
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/segments.spec.ts:382]: Hey @ViniLopes20, can you please declare the variables right before they have been used?

**Review by marcosapmf — CHANGES_REQUESTED** [review]

File: modules/test/playwright/tests/osb-faro-web/segments.spec.ts [hunk 2]
```diff
@@ -362,3 +368,264 @@ test('Search the Segment Profile Distribution', async ({apiHelpers, page}) => {
...
+
+	await test.step('Create the known individuals directly in the AC database', async () => {
+		await createIndividuals({
+			apiHelpers,
+			individuals: knownIndividual,
+		});
+	});
+
+	await test.step('Create an identity for an anonymous directly in the AC database', async () => {
+		const identities = [
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/segments.spec.ts:408]: I think you can inline this :)

File: modules/test/playwright/tests/osb-faro-web/segments.spec.ts [hunk 3]
```diff
@@ -362,3 +368,264 @@ test('Search the Segment Profile Distribution', async ({apiHelpers, page}) => {
...
+				createDate: date1.toISOString(),
+				id: anonymousIdentityID,
+			},
+		];
+
+		await apiHelpers.jsonWebServicesOSBAsah.createIdentities(identities);
+	});
+
+	await test.step('Create events for the anonymous and known individual to appear in AC', async () => {
+		const knownIndividualEvent = knownIndividual.map((individual) => ({
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/segments.spec.ts:419]: You probably can inline this on line 429 as well

File: modules/test/playwright/tests/osb-faro-web/segments.spec.ts [hunk 4]
```diff
@@ -362,3 +368,264 @@ test('Search the Segment Profile Distribution', async ({apiHelpers, page}) => {
...
+			eventId: 'pageViewed',
+			title: pageName,
+			userId: individual.id,
+		}));
+
+		await apiHelpers.jsonWebServicesOSBAsah.createEvents(
+			knownIndividualEvent
+		);
+
+		const anonymousIndividualEvents = [
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/segments.spec.ts:433]: Inline this on line 445

File: modules/test/playwright/tests/osb-faro-web/segments.spec.ts [hunk 5]
```diff
@@ -362,3 +368,264 @@ test('Search the Segment Profile Distribution', async ({apiHelpers, page}) => {
...
+			},
+		];
+
+		await apiHelpers.jsonWebServicesOSBAsah.createEvents(
+			anonymousIndividualEvents
+		);
+	});
+
+	await test.step('Create a session for the known individual', async () => {
+		const sessions = knownIndividual.map((individual) => ({
```
Comment (marcosapmf) [inline modules/test/playwright/tests/osb-faro-web/segments.spec.ts:451]: Inline this on line 459

**Comment by marcosapmf** [comment]
Great work @ViniLopes20 :tada: 
Just couple of SF adjustments that can be required on Brian's review :)

File: modules/test/playwright/tests/osb-faro-web/segments.spec.ts (hunk 2 above)
Comment (ViniLopes20) [reply modules/test/playwright/tests/osb-faro-web/segments.spec.ts:408]: Done

File: modules/test/playwright/tests/osb-faro-web/segments.spec.ts (hunk 3 above)
Comment (ViniLopes20) [reply modules/test/playwright/tests/osb-faro-web/segments.spec.ts:419]: Done

File: modules/test/playwright/tests/osb-faro-web/segments.spec.ts (hunk 5 above)
Comment (ViniLopes20) [reply modules/test/playwright/tests/osb-faro-web/segments.spec.ts:451]: Done

**Review by marcosapmf — APPROVED** [review]

**Comment by marcosapmf** [comment]
ci:forward

**Pull request closed without merging**